Library/Homebrew/resource.rb

```diff
@@ -67,7 +67,8 @@ def escaped_name
   end
 
   def download_name
-    name.nil? ? owner.name : "#{owner.name}--#{escaped_name}"
+    # Must not have "@" in download_name because it's a special character to Subversion
+    (name.nil? ? owner.name : "#{owner.name}--#{escaped_name}").gsub("@", "_AT_")
```

Comment:
`if name` would be a bit nicer (I know it's not your code). Would be good to reuse the `gsub`; perhaps a shared resource function?

Reply:
Refactored it to use an `if name`.